internal/render/pod_test.go

```diff
@@ -661,6 +664,166 @@ func TestCheckPhase(t *testing.T) {
+						statuses[i] = v1.ContainerStatus{
+							Name:  c.Name,
+							Ready: true,
+							State: v1.ContainerState{Running: &v1.ContainerStateRunning{}},
+						}
+					}
+					return statuses
+				}(),
+			},
+		}
+
+		// Convert to unstructured
+		obj, _ := runtime.DefaultUnstructuredConverter.ToUnstructured(pod)
+		return &unstructured.Unstructured{Object: obj}
+	}
+
+	always := v1.ContainerRestartPolicyAlways
+
+	tests := []struct {
+		name         string
+		pod          *unstructured.Unstructured
+		expectedGPUs string
+	}{
+		{
+			name: "no gpus",
+			pod: makePodWithGPU("test-pod", []v1.Container{
+				{Name: "app"},
+			}, nil),
+			expectedGPUs: "0",
+		},
+		{
+			name: "single container with nvidia gpu",
+			pod: makePodWithGPU("test-pod", []v1.Container{
+				{
+					Name: "app",
+					Resources: v1.ResourceRequirements{
+						Requests: v1.ResourceList{"nvidia.com/gpu": q("2")},
+					},
+				},
+			}, nil),
+			expectedGPUs: "2",
+		},
+		{
+			name: "multiple containers with different gpu vendors",
+			pod: makePodWithGPU("test-pod", []v1.Container{
+				{
+					Name: "app1",
+					Resources: v1.ResourceRequirements{
+						Requests: v1.ResourceList{"nvidia.com/gpu": q("1")},
+					},
+				},
+				{
+					Name: "app2",
+					Resources: v1.ResourceRequirements{
+						Requests: v1.ResourceList{"amd.com/gpu": q("2")},
+					},
+				},
+			}, nil),
+			expectedGPUs: "3",
+		},
+		{
+			name: "gpu in limits only",
+			pod: makePodWithGPU("test-pod", []v1.Container{
+				{
+					Name: "app",
+					Resources: v1.ResourceRequirements{
+						Limits: v1.ResourceList{"nvidia.com/gpu": q("4")},
+					},
+				},
+			}, nil),
+			expectedGPUs: "4",
+		},
+		{
+			name: "regular init container ignored",
+			pod: makePodWithGPU("test-pod",
+				[]v1.Container{
+					{
+						Name: "app",
+						Resources: v1.ResourceRequirements{
+							Requests: v1.ResourceList{"nvidia.com/gpu": q("1")},
+						},
+					},
+				},
+				[]v1.Container{
+					{
+						Name: "init",
+						Resources: v1.ResourceRequirements{
+							Requests: v1.ResourceList{"nvidia.com/gpu": q("10")},
+						},
+					},
+				},
+			),
+			expectedGPUs: "1",
+		},
+		{
+			name: "sidecar init container counted",
+			pod: makePodWithGPU("test-pod",
+				[]v1.Container{
+					{
+						Name: "app",
+						Resources: v1.ResourceRequirements{
+							Requests: v1.ResourceList{"nvidia.com/gpu": q("1")},
+						},
+					},
+				},
+				[]v1.Container{
+					{
+						Name:          "sidecar",
+						RestartPolicy: &always,
+						Resources: v1.ResourceRequirements{
+							Requests: v1.ResourceList{"nvidia.com/gpu": q("2")},
+						},
+					},
+				},
+			),
+			expectedGPUs: "3",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			pom := render.PodWithMetrics{
+				Raw: tt.pod,
+				MX:  nil, // No metrics needed for GPU test
+			}
+
+			po := render.NewPod()
+			r := model1.NewRow(30)
+			err := po.Render(&pom, "", &r)
```

Comment:
This ends up testing `render` which is ok, but best to single out `calculatePodGPUMetrics`